common/client-core/src/client/real_messages_control/real_traffic_stream.rs

```diff
@@ -271,7 +277,9 @@ where
         };
 
         if let Err(err) = self.mix_tx.send(vec![next_message]).await {
-            log::error!("Failed to send: {err}");
+            if !task_client.is_shutdown_poll() {
```

Comment:
would it perhaps be possible to move `TaskClient` to `Self` and call it that way instead? I think that's quite a good pattern to only log those errors if we're not in shutdown

Reply:
The problem is that we end up having two select branches with `&mut self`. (With `CancellationToken` this will not be a problem)

Reply:
Well I guess we can fork a clone for use in the select loop specifically